src/main/java/nl/workingtalent/bookrental/controller/BookController.java

```diff
@@ -193,14 +160,24 @@ public Book findBook(@PathVariable long id) {
 	}
 	
 	@GetMapping("book/copy/{id}")
-	public List<Copy> getAllCopiesById(@PathVariable long id) {
+	public List<Copy> getAvailableCopiesById(@PathVariable long id) {
 		
 		List<Copy> copies = copyRepo.findAll();
 		List<Copy> copiesOfBook = new ArrayList<Copy>();
 		
+		outerloop: 
 		for (Copy copy : copies) {
```

Comment:
Nu loop je denk ik over de gehele bieb? In ons geval nu niet erg, maar lijkt me vrij time consuming als de database wel heel groot is?

Reply:
Klopt, dit is mijn back-end manier van SQL queries. Ga ervan uit dat SQL queries exact hetzelfde doen, dus verwacht niet dat dit heel erg is. Het is log(n) op de big(O) schaal, wat niet perse heel slecht is. Als de tabel echt massief is zouden we met hashmaps/dictionaries kunnen werken, maar dat is zeker niet relevant tenzij we meer dan tienduizenden boeken hebben ;) Goed dat je het benoemt though, goede eigenschap om hier kritisch over te zijn, en actief te proberen te verbeteren!